### ISSUE TITLE
/reports/activity endpoint params not working in 6.3.1

### ISSUE THREAD

**Issue by squintfox** [grokability/snipe-it#14289]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

Since 6.3.1 (not in 6.3.0), some params for an API call to /reports/activity do not work.  Specifically action_type and order don't work.  Any value passed to action_type returns all results, order does not change (seems to always be desc).  item_type and id still work fine.

Unfortunately, I can't test this in the demo because there are no API keys.

### Reproduction steps

Make an API call to:
`https://server/api/v1/reports/activity?item_type=asset&item_id=1111&action_type=checkin from&order=asc`

### Expected behavior

Only events of action_type "checkin from" should be returned.  However, all events are returned.  Events should be returned with oldest first.  Events are returned newest first.

### Snipe-IT Version

v6.3.1 - build 12672 (master)

### Operating System

Ubuntu via Docker

### Web Server

Unsure, whatever is configured in docker

### PHP Version

8.1.2-1ubuntu2.14

**Comment by squintfox** [comment]
This commit is the source of the problem: https://github.com/snipe/snipe-it/commit/5e4a3379a954b35888133ca2e6f72b879de432db

My PHP isn't good enough to know exactly why, but I took that code out and it started working again.

**Comment by snipe** [comment]
The API explorer in the API docs includes an api key so you should be able to test it there

**Comment by squintfox** [comment]
@snipe Confirmed in develop with one new note.  The problem is specifically action_type.  If action_type is omitted, order works correctly.  If included, it does not.

![Screenshot 2024-02-19 132624](https://github.com/snipe/snipe-it/assets/55590532/669156bb-61b7-43d3-9c44-d6192a2b0c27)
![Screenshot 2024-02-19 132630](https://github.com/snipe/snipe-it/assets/55590532/c5d34bba-245d-40ca-9e67-1c84d9c13f4b)
![Screenshot 2024-02-19 132635](https://github.com/snipe/snipe-it/assets/55590532/39381729-8c7b-4736-b288-6f65f71f83aa)
![Screenshot 2024-02-19 132644](https://github.com/snipe/snipe-it/assets/55590532/ca395e1b-765e-47aa-af11-b512a8f3a688)
![Screenshot 2024-02-19 132931](https://github.com/snipe/snipe-it/assets/55590532/96688f72-00c9-41a7-81ed-a0c8f0232b40)

**Comment by snipe** [comment]
That's really helpful info, thank you! Looking into this now for you.

**Comment by snipe** [comment]
Just to clarify, you're not actually sending `checkin from` as a URL parameter, right? You're urlencoding it? Spaces should always be url encoded in URLs

**Comment by squintfox** [comment]
@snipe Yes, sorry.  Postman handles the encoding automatically.  It looks like this on the call:

![image](https://github.com/snipe/snipe-it/assets/55590532/a1d61138-ccf1-4803-8028-0652e5a4db95)

**Comment by squintfox** [comment]
@snipe I'm sure you're super busy, but if I can get you any more info on this one, let me know.

**Comment by squintfox** [comment]
@snipe I finally got some time to sit down and learn enough Laravel to understand what's going on here.  On it's face, the change in 6.3.1 seems pretty straight forward.  Both sides of the or clause work as intended and in a vacuum, it works fine as a whole when only item_id and item_type are passed.  Once you take a look at the SQL that Laravel is generating under the hood, it's pretty clear what's going wrong.

When you pass only item_id and item_type, the generated SQL query looks like this:

```select * from `action_logs` where (`item_id` = '86' and `item_type` = 'App\\Models\\Asset' or (`target_id` = '86' and `target_type` = 'App\\Models\\Asset'))```

When you add action_type as a 3rd parameter, you get this:

```select * from `action_logs` where (`item_id` = '86' and `item_type` = 'App\\Models\\Asset' or (`target_id` = '86' and `target_type` = 'App\\Models\\Asset') and `action_type` = 'checkin from')```

When you're adding more parameters after that or clause, it's not being restricted only to that one part of the statement.  Just a simple matter of some more parentheses needed.  I'm not a Laravel expert, but I worked up a PR that just uses an additional where statement to encapsulate that one part on it's own, so now you get:

```select * from `action_logs` where (`item_id` = '86' and `item_type` = 'App\\Models\\Asset' or (`target_id` = '86' and `target_type` = 'App\\Models\\Asset')) and `action_type` = 'checkin from'```

As an aside, I understand why the change was made so that the GUI display was more useful, but I also feel like it's a regression for the API users.  If I supply item_id, item_type and action_type, I'm expecting to get the actions of the item only (for example, the last time it was checked in).  Now I'm getting additional items that aren't immediately clear that they will be there.  I'm not sure if it merits a change, maybe just an update to the API documentation that you'll get related item_types to your item as well?
